From 8582002593e2fe5677974510a6c079721485f0de Mon Sep 17 00:00:00 2001
From: ColdWindScholar <3590361911@qq.com>
Date: Sat, 27 Jun 2026 15:57:04 +0800
Subject: [PATCH] fix booting

---
 loader/Loader.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/loader/Loader.cpp b/loader/Loader.cpp
index 08c6b6d..451f405 100644
--- a/loader/Loader.cpp
+++ b/loader/Loader.cpp
@@ -602,8 +602,10 @@ void vendorBpfLoader() {
         ALOGE("If this triggers randomly, you might be hitting some memory allocation "
               "problems or startup script race.");
         ALOGE("--- DO NOT EXPECT SYSTEM TO BOOT SUCCESSFULLY ---");
+        #if 0 //hacked
         sleep(20);
         exit(121);
+        #endif
     }
 
     const char* args[] = {"/apex/com.android.tethering/bin/netbpfload", "done", NULL};
-- 
2.53.0

