From 37e20b445a0a05fa8e01678d1266d10214ed8af4 Mon Sep 17 00:00:00 2001
From: Ziyang Zhou <ziyang.zhou@outlook.com>
Date: Sun, 9 May 2021 23:04:00 +0800
Subject: [PATCH] fix booting

---
 libs/ui/GraphicBufferMapper.cpp| 1 +
 1 files changed, 1 insertions(+), 1 deletion(-)
 
diff --git a/libs/ui/GraphicBufferMapper.cpp b/libs/ui/GraphicBufferMapper.cpp
index 91c760eac7..3370a261cf 100644
--- a/libs/ui/GraphicBufferMapper.cpp
+++ b/libs/ui/GraphicBufferMapper.cpp
@@ -54,7 +54,7 @@ using LockResult = GraphicBufferMapper::LockResult;
 ANDROID_SINGLETON_STATIC_INSTANCE( GraphicBufferMapper )
 
 static bool requireMapper4() {
-    return android_get_device_api_level() >= 36 && flags::require_gralloc4_or_newer();
+    return false;
 }
 
 void GraphicBufferMapper::preloadHal() {
 
---
2.53.0
