From c9dda8339b59cc3cfe2d4c4bbd2e423e5bfd6cff Mon Sep 17 00:00:00 2001
From: ColdWindScholar <3590361911@qq.com>
Date: Sat, 27 Jun 2026 15:26:37 +0800
Subject: [PATCH] ignore project quota error

---
 Utils.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Utils.cpp b/Utils.cpp
index a55a57e..0ccd444 100644
--- a/Utils.cpp
+++ b/Utils.cpp
@@ -262,7 +262,6 @@ int SetQuotaProjectId(const std::string& path, long projectId) {
             return 0; // swallow
         }
         PLOG(ERROR) << "Failed to set project id on " << path;
-        return ret;
     }
     return 0;
 }
-- 
2.53.0

