From 3422b65ac7f0ab8ce01e9a2c53fc3a0f208ad541 Mon Sep 17 00:00:00 2001
From: ColdWindScholar <3590361911@qq.com>
Date: Sat, 27 Jun 2026 13:06:13 +0800
Subject: [PATCH] ignore cgroup error

---
 ProcessList.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ProcessList.java b/ProcessList.java
index d8e2cfa..8ecb301 100644
--- a/ProcessList.java
+++ b/ProcessList.java
@@ -2694,7 +2694,7 @@ public final class ProcessList extends ProcessListInternal
                             if (res == -OsConstants.ESRCH) {
                                 Slog.e(ActivityManagerService.TAG, errorStr);
                             } else {
-                                throw new AssertionError(errorStr);
+                                Slog.e(ActivityManagerService.TAG, errorStr);
                             }
                         } else {
                             app.mProcessGroupCreated = true;
-- 
2.53.0

